From 0ff5ac617ea6abb8238d7c46065948a1f7eb2efa Mon Sep 17 00:00:00 2001
From: Nathan Muggli <nmuggli@google.com>
Date: Thu, 22 Oct 2026 09:32:20 -0600
Subject: [PATCH] Revert "Switch to forward decl of uint128 in //base/uuid.h"

This reverts commit 21c16c702dc23b339b9e70be148d88e972d7dcb5
(upstream commit 2e41c49f3039, r1724126, crrev.com/c/8614913).

ChromeOS uses system Abseil (dev-cpp/abseil-cpp) built with inline LTS
namespaces enabled (ABSL_OPTION_USE_INLINE_NAMESPACE=1, e.g.
absl::lts_20240116::uint128). Forward-declaring class uint128 directly
in namespace absl without ABSL_NAMESPACE_BEGIN/ABSL_NAMESPACE_END
declares a separate incomplete type ::absl::uint128, causing ambiguous
type lookup errors whenever <absl/numeric/int128.h> is also included.

patch-name: long-term-3300-Revert-Switch-to-forward-decl-of-uint128-in-base-uui.patch
Change-Id: Ife31a5f995d78ec3f7f563b23b5e561c232999fa
---
 base/uuid.h | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/base/uuid.h b/base/uuid.h
index 08dfa9ff26..73b022de14 100644
--- a/base/uuid.h
+++ b/base/uuid.h
@@ -16,18 +16,14 @@
 #include "base/compiler_specific.h"
 #include "base/containers/span.h"
 #include "build/robolectric_buildflags.h"
+#include "third_party/abseil-cpp/absl/numeric/int128.h"
 
 #if BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_ROBOLECTRIC)
 #include "base/android/jni_string.h"
 #include "base/system_jni/UUID_jni.h"
-#include "third_party/abseil-cpp/absl/numeric/int128.h"
 #include "third_party/jni_zero/jni_zero.h"
 #endif
 
-namespace absl {
-class uint128;
-}  // namespace absl
-
 namespace base {
 
 class BASE_EXPORT Uuid {
-- 
2.56.0.218.g1ec904b173-goog

